packages/dds/tree/src/test/memory/tree.spec.ts

```diff
@@ -82,10 +82,8 @@ describe("SharedTree memory usage", () => {
 		new (class implements IMemoryTestObject {
 			public readonly title = "Create empty SharedTree";
 
-			private sharedTree: TreeView<typeof RootNodeSchema> | undefined;
-
 			public async run(): Promise<void> {
-				this.sharedTree = createLocalSharedTree("testSharedTree");
+				createLocalSharedTree("testSharedTree");
```

Comment:
the whole point of these tests is to measure memory use. Since I can't tell what memory use its measuring (there is no before and after), it possible that this change to not keep the tree we just allocated in memory via this property changes what is being measure to not include the tree, defeating the point of this test.

Given you are trying to fix the fact we construct trees during test enumeration, this change seems irrelevant to that.

Reply:
Oh I see - JS GC might drop the reference before the test measures the memory. Good catch. I'll add a comment and underscore to the variable name so that future readers know it is intentional.

Reply:
Well, maybe this also explains why these tests are using classes. It's a convenient way to scope the lifetime of the sharedtree to the input object, rather than having to define it outside and close over it. There's nothing wrong with the latter, but I could see an argument for the class being more visually organized or something. I'm no longer bothered by it.

Reply:
Yep, that's exactly why I went with classes for this, as much as I found it super weird at the time. And still do today. Having variables defined outside and closed over seemed like it was screaming for hard-to-spot bugs related to patterns that could mess up with the memory measurements, like people reusing the shared variables across tests (like we do with most shared state in unit tests).

But very much open to alternative designs if anyone has good ideas and/or the time to implement them.